### PR TITLE
feat(xds)!: drop legacy Timeout consumption from xDS generation

### LINKED ISSUE #17213: feat(xds)!: drop Timeout from xDS generation
Part of #17203 — **Phase 2** (parallelizable after Phase 0).

## Motivation

Remove xDS consumption of legacy `Timeout` so it no longer affects generated Envoy config. Replaced by `MeshTimeout`. The API/resource definition stays until Phase 4 — resources remain accepted, stored, and KDS-synced during the migration window.

## Implementation information

- Drop the `Timeout` topology builder in `pkg/xds/topology/` (where applicable).
- Remove the `Timeout` field from the `MatchedPolicies` struct (`pkg/core/xds/matched_policies.go`) and its typed accessor in `pkg/xds/context/resources.go`.
- Remove population in `pkg/xds/sync/dataplane_proxy_builder.go` (`matchPolicies`) and `egress_proxy_builder.go`.
- Remove usage in the relevant `pkg/xds/generator/*_proxy_generator.go`.
- Remove any legacy back-compat shim for this policy in the corresponding `pkg/plugins/policies/MeshTimeout` plugin (e.g. `legacy_*_configurer.go`, `deprecated.go`).
- Update tests and golden files.

## Verification

Applying a `Timeout` produces no change in the Envoy config dump; `MeshTimeout` still works.

### PR DESCRIPTION
**Pull request by Automaat**
## Motivation

Phase 2 of #17203. Remove xDS consumption of legacy `Timeout` so it no longer affects generated Envoy config. The policy has been replaced by `MeshTimeout`. The API and resource definition remain until Phase 4, during which resources are still accepted, stored, and KDS-synced.

## Implementation information

- Drop the `Timeout` topology builder in `pkg/xds/topology/`
- Remove the `Timeout` field from the `MatchedPolicies` struct and its typed accessor in `pkg/xds/context/resources.go`
- Remove population in `pkg/xds/sync/dataplane_proxy_builder.go` and `egress_proxy_builder.go`
- Remove usage in proxy generators (`pkg/xds/generator/*_proxy_generator.go`)
- Remove legacy back-compat shims from the `MeshTimeout` plugin
- Update tests and golden files

Closes https://github.com/kumahq/kuma/issues/17213

_Generated by Sybra harness_

### CODE REVIEW

File: pkg/xds/envoy/virtualhosts/routes_configurer.go [hunk 1]
```diff
@@ -159,14 +160,8 @@ func (c RoutesConfigurer) hasExternal(clusters []envoy_common.Cluster) bool {
...
 	routeAction := &envoy_config_route_v3.RouteAction{}
-	if len(clusters) != 0 {
-		// Timeout can be configured only per outbound listener. So all clusters in the split
-		// must have the same timeout. That's why we can take the timeout from the first cluster.
-		if cluster, ok := clusters[0].(*envoy_common.ClusterImpl); ok {
-			routeAction.Timeout = util_proto.Duration(cluster.Timeout().GetHttp().GetRequestTimeout().AsDuration())
-		} else {
-			routeAction.Timeout = util_proto.Duration(0)
-		}
+	if c.ConfigureRouteTimeout && len(clusters) != 0 {
```
Comment (lukidzi) [inline pkg/xds/envoy/virtualhosts/routes_configurer.go:163]: Suggestion: The condition `c.ConfigureRouteTimeout && len(clusters) != 0` guards timeout=0s, but when len(clusters)==0 AND len(clusters)!=1, the code falls through to create an empty WeightedClusters. While upstream generators filter these out, defensively handling len(clusters)==0 explicitly would prevent future bugs.

File: pkg/xds/envoy/virtualhosts/configurer.go [hunk 2]
```diff
@@ -22,7 +22,16 @@ func DomainNames(domainNames ...string) VirtualHostBuilderOpt {
 func Routes(routes envoy_common.Routes) VirtualHostBuilderOpt {
 	return AddVirtualHostConfigurer(
 		&RoutesConfigurer{
-			Routes: routes,
+			Routes:                routes,
+			ConfigureRouteTimeout: true,
+		})
+}
+
+func OutboundRoutes(routes envoy_common.Routes) VirtualHostBuilderOpt {
```
Comment (lukidzi) [inline pkg/xds/envoy/virtualhosts/configurer.go:34]: Suggestion: The naming split Routes() vs OutboundRoutes() is correct but the semantics are subtle: Routes() sets timeout=0 (disable/infinite), OutboundRoutes() sets no timeout (lets MeshTimeout/Envoy default apply). A brief doc comment explaining this distinction would help future maintainers.

File: pkg/xds/envoy/virtualhosts/routes_configurer_test.go [hunk 3]
```diff
@@ -41,6 +46,18 @@ routes:
       prefix: "/"
     route:
       timeout: "0s"
+      cluster: backend`,
+		}),
+		Entry("routes without configured request timeout", testCase{
```
Comment (lukidzi) [inline pkg/xds/envoy/virtualhosts/routes_configurer_test.go:51]: Suggestion: Test coverage is minimal — only tests single-cluster routes. Consider adding tests for: (1) multiple weighted clusters with configureRouteTimeout=true/false, (2) edge case of empty clusters slice.

**Review by lukidzi — COMMENTED** [review]
No blocking issues found. Four non-blocking suggestions around documentation clarity and test coverage.

File: pkg/xds/envoy/virtualhosts/routes_configurer.go (hunk 1 above)
Comment (Automaat) [reply pkg/xds/envoy/virtualhosts/routes_configurer.go:163]: **Applied** — handled zero-cluster routes by leaving the cluster specifier unset instead of emitting empty `WeightedClusters` (`d10f0c29dc`).

_Generated by Sybra harness_

File: pkg/xds/envoy/virtualhosts/configurer.go (hunk 2 above)
Comment (Automaat) [reply pkg/xds/envoy/virtualhosts/configurer.go:34]: **Applied** — documented the `Routes()` vs `OutboundRoutes()` timeout semantics so the inbound legacy `0s` override and outbound policy-driven behavior are explicit (`d10f0c29dc`).

_Generated by Sybra harness_

File: pkg/xds/envoy/virtualhosts/routes_configurer_test.go (hunk 3 above)
Comment (Automaat) [reply pkg/xds/envoy/virtualhosts/routes_configurer_test.go:51]: **Applied** — added coverage for weighted routes with and without configured route timeouts, plus the zero-cluster edge case (`d10f0c29dc`).

_Generated by Sybra harness_

**Review by lukidzi — APPROVED** [review]
